client/seal_test.go

```diff
@@ -329,10 +331,130 @@ func BenchmarkSeal(b *testing.B) {
 				if err != nil {
 					b.Fatal(err)
 				}
-				if _, err = key.Unseal(blob, bm.cOpts); err != nil {
+				if _, err = key.Unseal(blob, bm.uOpts); err != nil {
 					b.Fatal(err)
 				}
 			}
 		})
 	}
 }
+func TestSealOpts(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+
+	emptySet := map[uint32]struct{}{}
+	srk, err := client.StorageRootKeyECC(rwc)
+	if err != nil {
+		t.Fatalf("failed to create SRK: %v", err)
+	}
+
+	opts := []struct {
+		name         string
+		current      tpm2.PCRSelection
+		target       *pb.PCRs
+		expectedPcrs map[uint32]struct{}
+	}{
+		{"CurrentEmpty-TargetNil", tpm2.PCRSelection{}, nil, emptySet},
+		{"CurrentEmpty7-TargetNil", tpm2.PCRSelection{}, nil, emptySet},
+		{"CurrentEmpty-TargetEmpty", tpm2.PCRSelection{}, &pb.PCRs{}, emptySet},
+		{"CurrentSHA1Empty-TargetSHA256Empty",
+			tpm2.PCRSelection{Hash: tpm2.AlgSHA1},
+			&pb.PCRs{Hash: pb.HashAlgo_SHA256},
+			emptySet},
+		{"CurrentSHA256Empty-TargetSHA1Empty",
+			tpm2.PCRSelection{Hash: tpm2.AlgSHA256},
+			&pb.PCRs{Hash: pb.HashAlgo_SHA1},
+			emptySet},
+		{"CurrentSHA2567-TargetSHA1Empty",
+			tpm2.PCRSelection{Hash: tpm2.AlgSHA256, PCRs: []int{7}},
+			&pb.PCRs{Hash: pb.HashAlgo_SHA1},
+			map[uint32]struct{}{7: {}}},
+		{"Current7-TargetPCR0,4",
+			tpm2.PCRSelection{Hash: tpm2.AlgSHA256, PCRs: []int{0, 7}},
+			&pb.PCRs{Hash: pb.HashAlgo_SHA256,
+				Pcrs: map[uint32][]byte{4: {0x00}}},
+			map[uint32]struct{}{
+				0: {},
+				4: {},
+				7: {},
+			}},
+	}
+
+	sliceToSet := func(a []uint32) map[uint32]struct{} {
+		ret := make(map[uint32]struct{})
+		for _, val := range a {
+			ret[val] = struct{}{}
+		}
+		return ret
+	}
+	for _, testcase := range opts {
+		t.Run(testcase.name, func(t *testing.T) {
+			sealed, err := srk.Seal([]byte("secretzz"),
+				client.SealOpts{Current: testcase.current, Target: testcase.target})
```

Comment:
Here and elsewhere in this file, this might be more readable as:
```go
opts := client.SealOpts{Current: testcase.current, Target: testcase.target}
sealed, err := srk.Seal([]byte("secretzz"), opts)
```

Reply:
oh looks like you prefer this to  the inline creation method. I have no preference so I'll do this and revert the changes I mentioned in cmd/seal.go above.